tarofreighter/chain_porter.go

```diff
@@ -1174,6 +1136,33 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage,
 		currentPkg.SendState = SendStateWaitTxConf
 		return &currentPkg, nil
 
+	// At this point, transaction broadcast is complete. We go on to wait
+	// for the transfer transaction to confirm on-chain.
+	case SendStateWaitTxConf:
+		err := p.waitForTransferTxConf(&currentPkg)
+		if err != nil {
+			return &currentPkg, err
```

Comment:
Should be `nil, err` in all those `if`s? Otherwise can just write `return &currentPkg, err` without the `if`.